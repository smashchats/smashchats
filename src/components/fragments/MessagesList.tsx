import React from "react";
import { View } from "react-native";

import { Box } from "@/src/components/design-system/Box";
import { Colors } from "@/src/constants/Colors";
import ProfileMessagesScreenText from "@/src/components/ProfileMessagesScreenText";
import ProfileMessagesScreenDate from "@/src/components/ProfileMessagesScreenDate";
import ProfileMessagesScreenMetadata from "@/src/components/ProfileMessagesScreenMetadata";
import { Message } from "@/src/app/profile/[user]/(tabs)/messages";
import { IM_CHAT_TEXT } from "@smashchats/library";

type MessagesListProps = {
    messages: Message[];
    paddingTop?: number;
};

export const MessagesList = ({
    messages,
    paddingTop = 0,
}: MessagesListProps) => {
    return (
        <Box marginHorizontal={10} paddingTop={paddingTop}>
            <Box width={"100%"} bg={Colors.background} paddingBottom={40}>
                {messages.map((m, idx) => {
                    let id = m.sha256 ?? `index-${idx}`;

                    switch (m.type) {
                        case IM_CHAT_TEXT:
                            return (
                                <ProfileMessagesScreenText
                                    key={`${m.type}-${id}`}
                                    message={m}
                                />
                            );
                        case "system-date":
                            return (
                                <ProfileMessagesScreenDate
                                    key={`${m.type}-${id}-index-${idx}`}
                                    date={m.date}
                                />
                            );
                        case "metadata":
                            return (
                                <ProfileMessagesScreenMetadata
                                    key={`${m.type}-${id}`}
                                    message={m}
                                />
                            );
                        case "profile":
                            return <View key={`${m.type}-${id}`} />;
                        case "profiles":
                            return (
                                <ProfileMessagesScreenText
                                    key={`${m.type}-${id}`}
                                    message={{
                                        ...m,
                                        content: "Several profiles [...]",
                                    }}
                                />
                            );
                        default:
                            return <Box key={`${m.type}-index-${idx}`} />;
                    }
                })}
            </Box>
        </Box>
    );
};

export default MessagesList;
