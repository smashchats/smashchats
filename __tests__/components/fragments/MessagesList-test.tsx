import { render } from "@testing-library/react-native";

import { MessagesList } from "@/src/components/fragments/MessagesList.jsx";

describe("MessagesList", () => {
    it("renders correctly", () => {
        const tree = render(<MessagesList messages={[]} />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("renders messages", () => {
        const messages = [
            {
                type: "message",
                content: "Hello, bob!",
                fromMe: true,
                sha256: "123",
                from: "alice",
                date: new Date(),
            },
            {
                type: "message",
                content: "Hello, alice!",
                fromMe: false,
                sha256: "124",
                from: "bob",
                date: new Date(),
            },
        ];
        const tree = render(<MessagesList messages={messages} />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
