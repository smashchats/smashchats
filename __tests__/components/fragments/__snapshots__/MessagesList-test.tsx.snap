// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessagesList renders correctly 1`] = `
<View
  style={
    {
      "backgroundColor": undefined,
      "height": undefined,
      "marginHorizontal": 10,
      "paddingTop": 0,
    }
  }
>
  <View
    style={
      {
        "backgroundColor": "#222222",
        "height": undefined,
        "paddingBottom": 40,
        "width": "100%",
      }
    }
  />
</View>
`;

exports[`MessagesList renders messages 1`] = `
<View
  style={
    {
      "backgroundColor": undefined,
      "height": undefined,
      "marginHorizontal": 10,
      "paddingTop": 0,
    }
  }
>
  <View
    style={
      {
        "backgroundColor": "#222222",
        "height": undefined,
        "paddingBottom": 40,
        "width": "100%",
      }
    }
  >
    <View
      style={
        {
          "backgroundColor": undefined,
          "height": undefined,
        }
      }
    />
    <View
      style={
        {
          "backgroundColor": undefined,
          "height": undefined,
        }
      }
    />
  </View>
</View>
`;
